packages/firestore/test/integration/api/database.test.ts

```diff
@@ -514,4 +515,55 @@ apiDescribe('Database', persistence => {
       return Promise.resolve();
     });
   });
+
+  asyncIt('can queue writes while offline', () => {
+    return withTestDb(persistence, db => {
+      const docRef = db.collection('rooms').doc();
+      const firestoreClient = (docRef.firestore as Firestore)._firestoreClient;
+
+      return firestoreClient
+        .disableNetwork()
+        .then(() => {
+          return Promise.all([
+            docRef.delete(),
+            firestoreClient.enableNetwork()
+          ]);
+        })
+        .then(() => Promise.resolve());
+    });
+  });
+
+  asyncIt('can get documents while offline', () => {
+    return withTestDb(persistence, db => {
+      const docRef = db.collection('rooms').doc();
+      const firestoreClient = (docRef.firestore as Firestore)._firestoreClient;
+
+      return firestoreClient
+        .disableNetwork()
+        .then(() => {
+          const promises = [];
+          let done: () => void;
+
+          promises.push(docRef.set({ a: 1 }));
+          promises.push(
+            new Promise(resolve => {
+              done = resolve;
```

Comment:
Could you use a Deferred (see other usages in this file) instead of this pattern?

Reply:
Yes, done (well, in the query test - see below)